### PR TITLE
Change local-node setup to lifecycle and boot up very early

### PR DESCRIPTION
**Pull request by jdillon**
Helps expose local node-id to other services, like for orient-db:

https://gist.github.com/jdillon/7357667f664441524981

Removes ability to "reset" which isn't a good idea anyways.  Capability still exists, but serves mainly to expose information, pending if we can make this never able to disable or remove.  May still remove completly and provide details somewhere else.

May merge this into shared-hz branch, as its not critical to master atm.

### CODE REVIEW

File: components/nexus-core/src/main/java/org/sonatype/nexus/internal/node/IdentityCapability.java [hunk 1]
```diff
@@ -67,61 +56,12 @@ protected IdentityCapabilityConfiguration createConfig(final Map<String, String>
...
-
-  @Override
-  protected void onRemove(final IdentityCapabilityConfiguration config) throws Exception {
-    if (keyStoreManager.isKeyPairInitialized()) {
-      log.debug("Clearing identity keys");
-      keyStoreManager.removePrivateKey();
-    }
-    fingerprint = null;
-  }
+  // FIXME: This does not actually work, will have to add some sort of hook/condition/magic
```
Comment (kellyrob99) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/node/IdentityCapability.java]: ticket to cover this?

Comment (jdillon) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/node/IdentityCapability.java]: No, outstanding question to @adreghiciu about it though in email.

Comment (sonatype-zion) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/node/IdentityCapability.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)

File: components/nexus-core/src/main/java/org/sonatype/nexus/internal/node/IdentityCapability.java [hunk 2]
```diff
@@ -67,61 +56,12 @@ protected IdentityCapabilityConfiguration createConfig(final Map<String, String>
...
+  //@Override
+  //protected void onRemove(final IdentityCapabilityConfiguration config) throws Exception {
```
Comment (sonatype-zion) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/node/IdentityCapability.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: components/nexus-core/src/main/java/org/sonatype/nexus/internal/node/IdentityCapability.java [hunk 3]
```diff
@@ -67,61 +56,12 @@ protected IdentityCapabilityConfiguration createConfig(final Map<String, String>
...
+  //  // HACK: until we have a condition to prevent this
+  //  throw new IllegalStateException("Capability can not be removed");
```
Comment (sonatype-zion) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/node/IdentityCapability.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

File: components/nexus-common/src/main/java/org/sonatype/nexus/common/node/LocalNodeAccess.java [hunk 4]
```diff
@@ -12,6 +12,12 @@
  */
 package org.sonatype.nexus.common.node;
 
+import java.security.cert.Certificate;
+
+import org.sonatype.sisu.goodies.lifecycle.Lifecycle;
+
+// TODO: rename to LocalNodeIdentity ?
```
Comment (sonatype-zion) [inline components/nexus-common/src/main/java/org/sonatype/nexus/common/node/LocalNodeAccess.java]: INFO Complete the task associated to this TODO comment. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

**Comment by sonatype-zion** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 3 major
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**